WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift

```diff
@@ -218,6 +241,29 @@ private extension OrdersRootViewController {
             self?.configureNavigationButtons(isOrderCreationExperimentalToggleEnabled: isOrderCreationEnabled)
         }
     }
+
+    /// Connect hooks on `ResultsController` and query cached data.
+    /// This is useful for stay up to date with the remote statuses, resetting the filters if one of the local status filters was deleted remotely.
+    ///
+    func configureStatusResultsController() {
+        statusResultsController.onDidChangeObject = { [weak self] (updatedOrdersStatus, _, _, _) in
+            guard let self = self else { return }
+            self.resetFiltersIfAnyStatusFilterIsNoMoreExisting(orderStatuses: self.statusResultsController.fetchedObjects)
+        }
+
+        try? statusResultsController.performFetch()
+        self.resetFiltersIfAnyStatusFilterIsNoMoreExisting(orderStatuses: self.statusResultsController.fetchedObjects)
+    }
+
+    /// If the current applied status filters does not match the existing status filters fetched from API, we reset them.
+    ///
+    func resetFiltersIfAnyStatusFilterIsNoMoreExisting(orderStatuses: [OrderStatus]) {
+        for orderStatus in self.statusResultsController.fetchedObjects {
+            if self.filters.orderStatus?.contains(orderStatus.status) == false {
+                self.clearFilters()
```

Comment:
nit: `self` might not be necessary and it can use the given `orderStatuses`?

```suggestion
        for orderStatus in orderStatuses {
            if filters.orderStatus?.contains(orderStatus.status) == false {
                clearFilters()
```

Reply:
Is it possible to just check if `orderStatuses` contain `filters.orderStatus`?

Reply:
`filters.orderStatus` it's an array, so we should iterate over the elements